ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/DetailsPageHeader.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import {
+    Flex,
+    FlexItem,
+    Label,
+    LabelGroup,
+    LabelProps,
+    Skeleton,
+    Text,
+    Title,
+} from '@patternfly/react-core';
+
+export type PageHeaderLabel = {
+    text: string;
+    icon: React.ReactNode;
+    color: LabelProps['color'];
+};
+
+export type DetailsPageHeaderProps = {
+    isLoading: boolean;
+    name: string;
+    labels?: PageHeaderLabel[];
+    summary?: string;
+    nameScreenReaderText: string;
+    metadataScreenReaderText: string;
+};
+
+const MAX_NUM_LABELS = 7;
+
+function DetailsPageHeader({
+    isLoading,
+    name,
+    labels,
+    summary,
+    nameScreenReaderText,
+    metadataScreenReaderText,
+}: DetailsPageHeaderProps) {
+    if (isLoading) {
```

Comment:
I wonder if doing the `isLoading` checks at the content level would make more sense. That might make future layout changes easier although it would result in multiple `isLoading` checks.

Something like this:
```
{summary && (
    <FlexItem>
        {isLoading ? (
            <Skeleton screenreaderText={metadataScreenReaderText} />
        ) : (
            <Text>summary</Text>
        )}
    </FlexItem>
)}
```
This way we would get a summary skeleton only if summary is included. It also would allow you to change `FlexItem` without having to make layout changes in two places.

Reply:
That is a great question and suggestion. We should know whether we need to include a summary or not (same with labels, etc.). If we don't, then you're right that showing a skeleton for the summary wouldn't be needed. Also, the benefit of not having to make changes in two places is quite helpful. I don't think there should be an issue doing it on the content level. Thanks for the suggestion!